modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPathAttributeGenerator.cs

```diff
@@ -93,8 +95,16 @@ IEnumerable<ClassDeclarationSyntax> classDeclarations
                 if (attributes.Length != 0)
                     attributes.Append("\n");
 
+                string scriptPath = RelativeToDir(cds.SyntaxTree.FilePath, godotProjectDir);
+                if (usedPaths.Contains(scriptPath))
```

Comment:
```suggestion
                if (!usedPaths.Add(scriptPath))
```

`HashSet.Add()` returns `false` when the element is already present.